src/components/Navigation.tsx

```diff
@@ -0,0 +1,83 @@
+import styled from "@emotion/styled";
+import LeftArrow from "@/components/icons/LeftArrow";
+import Profile from "@/components/icons/Profile";
+import { useLocation, useNavigate } from "react-router-dom";
+import { ROUTE_PATH } from "@/App";
+import Button from "@/components/common/Button";
+
+const Navigation = () => {
+  const location = useLocation();
+  const navigate = useNavigate();
+  const handleBackClick = () => {
+    if (window.history.length > 1) {
+      navigate(-1);
+    } else {
+      navigate(ROUTE_PATH.HOME);
+    }
+  };
+  const handleHomeClick = () => {
+    navigate(ROUTE_PATH.HOME);
+  };
+  const handleLoginClick = () => {
+    let path: string = `${ROUTE_PATH.LOGIN}`;
+    if ((Object.values(ROUTE_PATH) as string[]).includes(location.pathname)) {
+      path = path + `?redirect=${location.pathname}`;
+    }
+    navigate(path);
+  };
```

Comment:
이 함수 하는 일로 네이밍을 해주면 어떨까요?
```suggestion
  const goBack = () => {
    if (window.history.length > 1) {
      navigate(-1);
    } else {
      navigate(ROUTE_PATH.HOME);
    }
  };
  const goHome = () => {
    navigate(ROUTE_PATH.HOME);
  };
  const goLogin = () => {
    let path: string = `${ROUTE_PATH.LOGIN}`;
    if ((Object.values(ROUTE_PATH) as string[]).includes(location.pathname)) {
      path = path + `?redirect=${location.pathname}`;
    }
    navigate(path);
  };
```
요로코롬...